compiler/K_lift.fx

```diff
@@ -416,15 +421,36 @@ fun lift_all(kmods: kmodule_t list)
 
     fun fold_defcl_ktyp_(t: ktyp_t, loc: loc_t, callb: k_fold_callb_t) {}
 
+    var functions_met_as_vals = empty_id_hashset(1)
+    var declosured_functions = empty_id_hashset(1)
+
+    fun fold_defcl_atom_(a: atom_t, loc: loc_t, callb: k_fold_callb_t): void  = 
+        match a {
+        | AtomId(n) => 
+            match kinfo_(n, loc) {
+            | KFun (ref {kf_name}) => 
+                functions_met_as_vals.add(kf_name)
+            | _ => {}
+            }
+        | _ => {}
+        }
+
     // form a closure for each function that has free variables (mutable or not)
     fun fold_defcl_kexp_(e: kexp_t, callb: k_fold_callb_t)
     {
-        fold_kexp(e, callb)
         match e {
         | KDefFun kf =>
-            val {kf_name, kf_params, kf_rt, kf_closure, kf_flags, kf_scope, kf_loc} = *kf
+            val {kf_name, kf_params, kf_rt, kf_body, kf_closure, kf_flags, kf_scope, kf_loc} = *kf
+            val {kci_wrap_f} = kf_closure
+            //We cannot use fold_kexp directly on e, because we don't want to count function id in functions_met_as_vals.
+            //Thus, folding different details separately.
+            for arg <- kf_params {check_n_fold_id(arg, kf_loc, callb)}
```

Comment:
because formal function parameters/arguments are not functions, this line `for arg <- kf_params {...}` is unnecessary

Reply:
done